File: go/src/internal/events/events.go
package events

const (
	// Deposit lifecycle events (published by create_new_deposit after persisting)
	Source               = "interview.mock.deposits"
	DepositRequestedType = "deposit-requested"

	// Upstream events that trigger a new deposit
	PendingTransactionSource = "interview.mock.transactions"
	PendingTransactionType   = "transaction-pending"
)

type DepositRequestedEvent struct {
	Id            string `json:"id"`
	TransactionId string `json:"transactionId"`
	CustomerId    string `json:"customerId"`
}

type PendingTransactionEvent struct {
	TransactionId string      `json:"transactionId"`
	CustomerId    string      `json:"customerId"`
	Amount        string      `json:"amount"`
	Name          string      `json:"name"`
	Document      string      `json:"document"`
	BankAccount   BankAccount `json:"bankAccount"`
}

type BankAccount struct {
	Bank    string `json:"bank"`
	Branch  string `json:"branch"`
	Account string `json:"account"`
}

File: python/src/internal/events/events.py
from dataclasses import dataclass
from typing import Any

# Deposit lifecycle events (published by stream_consumer on DDB transitions)
SOURCE = "interview.mock.deposits"
DEPOSIT_REQUESTED_TYPE = "deposit-requested"

# Upstream events that trigger a new deposit
PENDING_TRANSACTION_SOURCE = "interview.mock.transactions"
PENDING_TRANSACTION_TYPE = "transaction-pending"


@dataclass
class BankAccount:
    bank: str
    branch: str
    account: str

    @classmethod
    def from_json(cls, raw: dict[str, Any]) -> "BankAccount":
        return cls(bank=raw["bank"], branch=raw["branch"], account=raw["account"])


@dataclass
class PendingTransactionEvent:
    transaction_id: str
    customer_id: str
    amount: str
    name: str
    document: str
    bank_account: BankAccount

    @classmethod
    def from_json(cls, raw: dict[str, Any]) -> "PendingTransactionEvent":
        return cls(
            transaction_id=raw["transactionId"],
            customer_id=raw["customerId"],
            amount=raw["amount"],
            name=raw["name"],
            document=raw["document"],
            bank_account=BankAccount.from_json(raw["bankAccount"]),
        )


@dataclass
class DepositRequestedEvent:
    id: str
    transaction_id: str
    customer_id: str

    @classmethod
    def from_json(cls, raw: dict[str, Any]) -> "DepositRequestedEvent":
        return cls(
            id=raw["id"],
            transaction_id=raw["transactionId"],
            customer_id=raw["customerId"],
        )
